packages/terminal/src/browser/search/terminal-search-widget.tsx

```diff
@@ -0,0 +1,182 @@
+            <button title='Next match' className='search-elem' onClick={this.onNextButtonClicked}>&#187;</button>
+            <button title='Close' className='search-elem close' onClick={this.hide}></button>
+       </div>;
+    }
+
+    onSearchInputFocus(): void {
+        if (this.searchBox) {
+            this.searchBox.classList.add('option-enabled');
+        }
+    }
+
+    onSearchInputBlur(): void {
+        if (this.searchBox) {
+            this.searchBox.classList.remove('option-enabled');
+        }
+    }
+
+    private onCaseSensitiveOptionClicked(event: React.MouseEvent<HTMLSpanElement>): void {
+        this.searchOptions.caseSensitive = !this.searchOptions.caseSensitive;
+        this.updateSearchInputBox(this.searchOptions.caseSensitive, event.currentTarget);
+    }
+
+    private onWroleWordOptionClicked(event: React.MouseEvent<HTMLSpanElement>): void {
+        this.searchOptions.wholeWord = !this.searchOptions.wholeWord;
+        this.updateSearchInputBox(this.searchOptions.wholeWord, event.currentTarget);
+    }
+
+    private onRegexOptionClicked(event: React.MouseEvent<HTMLSpanElement>): void {
+        this.searchOptions.regex = !this.searchOptions.regex;
+        this.updateSearchInputBox(this.searchOptions.regex, event.currentTarget);
+    }
+
+    private updateSearchInputBox(enable: boolean, optionElement: HTMLSpanElement): void {
+        if (enable) {
+            optionElement.classList.add('option-enabled');
+        } else {
+            optionElement.classList.remove('option-enabled');
+        }
+        this.searchInput!.focus();
+    }
+
+    private onInputChanged(event: React.KeyboardEvent): void {
+        // move to previous search result on `Shift + Enter`
+        if (event && event.shiftKey && event.keyCode === Key.ENTER.keyCode) {
+            this.search(false, 'previous');
+            return;
+        }
+        // move to next search result on `Enter`
+        if (event && event.keyCode === Key.ENTER.keyCode) {
+            this.search(false, 'next');
+            return;
+        }
+
+        this.search(true, 'next');
+    }
+
+    search(incremental: boolean, searchDirection: 'next' | 'previous'): void {
+        if (this.searchInput) {
+            this.searchOptions.incremental = incremental;
+            const searchText = this.searchInput.value;
+
+            if (searchDirection === 'next') {
+                findNext(this.terminal, searchText, this.searchOptions);
+            }
+
+            if (searchDirection === 'previous') {
+                findPrevious(this.terminal, searchText, this.searchOptions);
+            }
+        }
+    }
+
+    private onNextButtonClicked(): void {
+        this.search(false, 'next');
+    }
+
+    private onPreviousButtonClicked(): void {
+        this.search(false, 'previous');
+    }
+
+    hide(): void {
+        super.hide();
+        this.terminal.focus();
+    }
+
+    show(): void {
+        super.show();
+        if (this.searchInput) {
```

Comment:
it should be implemented in `onAfterShow` lifecycle callback

Reply:
Done